tests/integration/targets/consul/tasks/consul_agent_service.yml

```diff
@@ -0,0 +1,176 @@
+      - result.service.Tags[1] == 'new_tag'
+      - result.service.Meta.nginx_version is defined
+      - result.service.Meta.nginx_version == '1.25.3'
+      - result.service.Meta.nginx is defined
+      - result.service.Meta.nginx == '1.25.3'
+      - result.service.ContentHash is defined
+
+- name: Add a check for service {{ nginx_service.ID }}
+  community.general.consul_agent_check:
+    name: nginx_check
+    id: nginx_check
+    interval: 30s
+    http: http://localhost:80/morestatus
+    notes: "Nginx Check"
+    token: "{{ consul_management_token }}"
+    service_id: "{{ nginx_service.ID }}"
+  register: result
+
+- assert:
+    that:
+      - result is changed
+      - result.check is defined
+      - result.check.CheckID == 'nginx_check'
+      - result.check.ServiceID == 'nginx'
+      - result.check.Interval == '30s'
+      - result.check.Type == 'http'
+      - result.check.Notes == 'Nginx Check'
+
+
+- name: "Update a check with name nginx_check for service {{ nginx_service.ID }}"
+  community.general.consul_agent_check:
+    name: nginx_check
+    id: nginx_check
+    interval: 60s
+    http: https://local-host.de/morestatus
+    notes: "New Nginx Check"
+    token: "{{ consul_management_token }}"
+    service_id: "{{ nginx_service.ID }}"
+  register: result
+
+- assert:
+    that:
+      - result is changed
+      - result.check is defined
+      - result.check.CheckID == 'nginx_check'
+      - result.check.ServiceID == 'nginx'
+      - result.check.Interval == '1m0s'
+      - result.check.Type == 'http'
+      - result.check.Notes == 'New Nginx Check'
+
+- name: "Remove check with Id: nginx_check"
+  community.general.consul_agent_check:
+    id: nginx_check
+    state: absent
+    service_id: "{{ nginx_service.ID }}"
+    token: "{{ consul_management_token }}"
+  register: result
+
+- assert:
+    that:
+      - result is changed
+      - result is not failed
+      - result.operation == 'remove'
+
+- name: "Remove a service with Id: {{ nginx_service.ID }}"
+  community.general.consul_agent_service:
+    id: "{{ nginx_service.ID }}"
+    state: absent
+    token: "{{ consul_management_token }}"
+  register: result
+
+- assert:
+    that:
+      - result is changed
+      - result is not failed
+      - result.operation == 'remove'
+
+
+- name: register nginx with a tcp check
+  community.general.consul_agent_service:
+    name: nginx
+    service_port: 80
+    checks:
+      - name: nginx-check2
+        interval: 60s
+        tcp: localhost:80
+    token: "{{ consul_management_token }}"
+  register: result
+
+- assert:
+    that:
+      - result is changed
+      - result.service.ID is defined
+      - result.service.Service == 'nginx'
+
+- name: Register nginx with an http check
+  community.general.consul_agent_service:
+    name: nginx
+    service_port: 80
+    checks:
+      - name: nginx-check2
+        interval: 60s
+        http: http://localhost:80/status
+    token: "{{ consul_management_token }}"
```

Comment:
If you configure action groups as mentioned above, the token would be applied automatically here.